hack/lib/etcd.sh

```diff
@@ -103,7 +103,9 @@ kube::etcd::cleanup() {
 kube::etcd::install() {
   (
     cd "${KUBE_ROOT}/third_party"
-    if [[ $(readlink etcd) == etcd-v${ETCD_VERSION}-* ]]; then
```

Comment:
Let's do something like this:
```
local os
os=$(uname | tr "[:upper:]" "[:lower:]")
if [[ $(readlink etcd) == etcd-v${ETCD_VERSION}-${os}-* ]]; then
  return  # already installed
fi
if [[ ${os} == "darwin" ]]; then
```

That way we verify the link matches the OS, and if not, redownload.

Reply:
Thanks for your precise suggestion. I have updated accordingly along with the addition of `rm "${download_file}"` for Linux block.